hooks/pre_gen_project.py

```diff
@@ -1,10 +1,86 @@
+import requests
+from typing import List
 import re
 import sys
 
+# Attempt to import BeautifulSoup; handle if it's not installed
+try:
+    from bs4 import BeautifulSoup
+    BS_AVAILABLE = True
+except ImportError:
+    BS_AVAILABLE = False
+
 MODULE_REGEX = r'^[_a-zA-Z][_a-zA-Z0-9]+$'
 MODULE_NAME = '{{ cookiecutter.project_name_underscored }}'
 PROJECT_REGEX = r'^[a-zA-Z0-9]+(?:-[a-zA-Z0-9]+)*$'
 PROJECT_NAME = '{{ cookiecutter.project_name_dashed }}'
+PYTHON_VERSION = '{{ cookiecutter.python_version }}'
+
+
+# Hardcoded list of Python versions as a fallback
+HARDCODED_PYTHON_VERSIONS: List[str] = [
+  "3.14",
+  "3.13",
+  "3.12",
+  "3.11",
+  "3.10",
+  "3.9",
+  "3.8"
+]
+
+def get_available_python_versions() -> List[str]:
```

Comment:
Parsing HTML is not very robust. Do we really need that?

There are plenty of alternatives:
- Just using hardcoded values (and updating every 17 months)
- Relying on Python's [very stable release schedule](https://devguide.python.org/versions/), as per [PEP 602](https://peps.python.org/pep-0602/)
- Making use of [trove classifiers](https://github.com/pypa/trove-classifiers), e.g., `Programming Language :: Python :: 3.12`, see [here](https://github.com/pypa/trove-classifiers/blob/main/src/trove_classifiers/__init__.py)
- Use `versions-manifest.json` from `actions/https://github.com/actions/python-versions`
- Try to install the indicated Python version and catch the error
- Just let it fail

You can also do a mix of multiple of these, depending what you actually use the Python version for. For example, if you need to install the Python version via Poetry, your best validation is to just try installing it via Poetry. If you (also) need to make sure the Python version is available on GitHub Actions, check if it's available in the `versions-manifest.json`. And so on. This is not only more robust in terms of the implementation, but also a more meaningful validation.

Reply:
Yep this makes much more sense.